app/components/bookings/edit/edit-label.component.ts

```diff
@@ -0,0 +1,95 @@
+import {Component, Input} from '@angular/core';
+import {IONIC_DIRECTIVES, NavController, Toast} from "ionic-angular/index";
+import {Booking} from "../../../models/booking/booking";
+import {HoursToWorkdaysPipe} from "../../../pipes/hours-to-workdays.pipe";
+import {BookingService} from "../../../services/bookings/booking.service";
+
+@Component({
+  selector: 'edit-label',
+  directives: [IONIC_DIRECTIVES],
+  templateUrl: 'build/components/bookings/edit/edit-label.component.html',
+  pipes: [HoursToWorkdaysPipe]
+})
+
+export class EditLabel {
+
+  private _is_edit_mode: boolean = false;
+  private _rememberValue: String= "";
+
+  @Input()
+  public _text: String;
+  @Input()
+  public booking: Booking;
+
+  constructor( private _navController: NavController,
+               private _bookingService: BookingService) {
+  }
+
+
+  private _focus():void{
+
+  }
+
+  private _update_booking ():void{
+
+    this.booking.description = String(this._text);
+    //TODO: test on valid backend istallation
+
+    // once got this error in backend:
+
+    /*
+    *POST /api/projects/3/bookings/
+     HTTP 400 Bad Request
+     {
+     "detail": "JSON parse error - Extra data: line 20 column 6 - line 20 column 7 (char 614 - 615)"
+     }
+
+
+     File "...WBS-Project-timetracker-backend\wbs_timetracker\api\bookings\serializers.py", line 33, in create
+     if workpackage.is_toplevel_wp:
+     AttributeError: 'OrderedDict' object has no attribute 'is_toplevel_wp
+    *
+    * */
+
+    this._showToast();
+    this._bookingService.update(this.booking).subscribe((returnedBooking: Booking) => {
+      console.log("returned booking after update: " + returnedBooking);
```

Comment:
Debugging messages should removed.
The Toast should definitely not be displayed before the actual updating happens, but rather when you have successfully updated the booking. And there should definitely should be some sort of feedback to the user, when it didn't work (which could happen quite frequently, since we are on a mobile device and having an internet connection is unfortunately not always the case). Also in that case, what happens to the changes? If they are not reverted, the user might be confused, since he sees the changes, but they haven't been saved to the backend. As far as I can tell, you already have the initial value cached, why not just reset it, when shit goes wrong?